tests/tests_geomstats/test_full_rank_correlation_matrices.py

```diff
@@ -1,148 +1,237 @@
+        bundle = self.space(n)
+        horizontal_vec = bundle.horizontal_projection(vec, mat)
         inverse = GeneralLinear.inverse(mat)
         product_1 = Matrices.mul(horizontal_vec, inverse)
         product_2 = Matrices.mul(inverse, horizontal_vec)
-        is_horizontal = self.base.is_tangent(
-            product_1 + product_2, mat, atol=gs.atol * 10
+        is_horizontal = gs.all(
+            bundle.base.is_tangent(product_1 + product_2, mat, atol=gs.atol * 10)
         )
-        self.assertTrue(is_horizontal)
-
-    def test_horizontal_lift_and_tangent_riemannian_submersion(self):
-        mat = self.base.random_point()
-        vec = self.base.random_point()
-        tangent_vec = self.base.to_tangent(vec, mat)
-        horizontal = self.bundle.horizontal_lift(tangent_vec, base_point=mat)
-        result = self.bundle.tangent_riemannian_submersion(horizontal, mat)
+        self.assertAllClose(is_horizontal, gs.array(True))
+
+    def test_horizontal_lift_is_horizontal(self, n, tangent_vec, mat):
+        bundle = self.space(n)
+        lift = bundle.horizontal_lift(gs.array(tangent_vec), gs.array(mat))
+        result = gs.all(bundle.is_horizontal(lift, gs.array(mat)))
+        self.assertAllClose(result, gs.array(True))
+
+    def test_vertical_projection_is_vertical(self, n, tangent_vec, mat):
+        bundle = self.space(n)
+        proj = bundle.vertical_projection(gs.array(tangent_vec), gs.array(mat))
+        result = gs.all(bundle.is_vertical(proj, gs.array(mat)))
+        self.assertAllClose(result, gs.array(True))
+
+    @autograd_tf_and_torch_only
+    def test_align_is_horizontal(self, n, point_a, point_b):
+        bundle = self.space(n)
+        aligned = bundle.align(point_a, point_b, tol=1e-10)
+        log = bundle.ambient_metric.log(aligned, point_b)
+        result = bundle.is_horizontal(log, point_b, atol=gs.atol * 10000)
+        self.assertAllClose(result, gs.array(True))
+
+    def test_horizontal_lift_and_tangent_riemannian_submersion(
+        self, n, tangent_vec, mat
+    ):
+        bundle = self.space(n)
+        horizontal = bundle.horizontal_lift(gs.array(tangent_vec), gs.array(mat))
+        result = bundle.tangent_riemannian_submersion(horizontal, gs.array(mat))
         self.assertAllClose(result, tangent_vec)
 
-    def test_is_horizontal(self):
-        mat = self.base.random_point()
-        vec = self.base.random_point()
-        tangent_vec = self.base.to_tangent(vec, mat)
-        horizontal = self.bundle.horizontal_lift(tangent_vec, mat)
-        result = self.bundle.is_horizontal(horizontal, mat)
-        self.assertTrue(result)
-
-    def test_is_vertical(self):
-        mat = self.bundle.random_point()
-        tangent_vec = self.bundle.random_point()
-        vertical = self.bundle.vertical_projection(tangent_vec, mat)
-        result = self.bundle.is_vertical(vertical, mat)
-        self.assertTrue(result)
-
-    def test_inner_product(self):
```

Comment:
this is missing

Reply:
yeah. I removed.  this is not ideal way of testing. this is making sure two different implementations of innerproduct match. what if the test_implementation is wrong.